selfservice/flow/settings/handler.go

```diff
@@ -298,7 +301,13 @@ func (h *Handler) createBrowserSettingsFlow(w http.ResponseWriter, r *http.Reque
 		return
 	}
 
-	if err := h.d.SessionManager().DoesSessionSatisfy(r, s, h.d.Config().SelfServiceSettingsRequiredAAL(r.Context())); err != nil {
+	var managerOptions []session.ManagerOptions
+	requestURL := x.RequestURL(r)
+	if requestURL.Query().Get("return_to") != "" {
+		managerOptions = append(managerOptions, session.WithRequestURL(requestURL.String()))
```

Comment:
This is where the magic is happening right? The URL is basically:

```
/settings/browser?return_to=/login_browser?return_to=https://source"
```

Reply:
it's in multiple places in the settings handler. 

Since we have a create, fetch and update flow, all three need to pass along a request URL to the `DoesSessionSatisfy` method which then appends the request URL to the `/login?aal=aal2` URL. 

The true magic is here:
https://github.com/ory/kratos/pull/3263/files#diff-a50a45cf2bc8e265082b551b7fb148ba52b47e6e6e1e0266d388bbd7955e20ccR338